org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/JavaFormatterGrammarTest.java

```diff
@@ -508,7 +508,33 @@ public void beeLanguage() {
 				);
 		// @formatter:on
 	}
-	
+
+	@Test
+	public void xtextDocumentModifyTest_testCommentsNotDuplicated() {
+		// @formatter:off
+		assertFormattedGrammar(
+				"generate foo \"http://foo.net/foo\"" + NL +
+				"Foo: // comment in Foo " + NL +
+				"// comment before Assignment" + NL +
+				"  bars+= /* comment in assignment */ Bar // comment after assignment" + NL +
+				"// comment before keywod" + NL +
+				"'foo';" + NL +
+				"Bar: 'bar';"
+				,
+				"generate foo \"http://foo.net/foo\"" + NL +
+				""  + NL +
+				"Foo: // comment in Foo " + NL +
+				"// comment before Assignment" + NL +
```

Comment:
will would prefer to see tabs inside the strings
than as constant

Reply:
Done.